### PR TITLE
Run Sphinx' linkcheck on Travis + fix incorrect links

### PR DESCRIPTION
**Pull request by rutsky**
This is reopened #2035 

This should fix [issue 3478](http://trac.buildbot.net/ticket/3478).

This is WIP: currently link checker is enabled to get list of issues that we have.
Feel free to take and resolve this issue.

### CODE REVIEW

**Comment by rutsky** [comment]
Currently broken links:

```
make -C master/docs SPHINXOPTS=-q linkcheck

make: Entering directory `/home/travis/build/buildbot/buildbot/master/docs'

sphinx-build -b linkcheck -d _build/doctrees  -q . _build/linkcheck

/home/travis/build/buildbot/buildbot/master/docs/developer/cls-forcesched.rst:70: WARNING: broken link: https://github.com/buildbot/buildbot/blob/master/master/buildbot/status/web/template/forms.html

/home/travis/build/buildbot/buildbot/master/docs/developer/data.rst:195: WARNING: broken link: https://github.com/buildbot/buildbot/blob/master/master/buildbot/data/pubs.py

/home/travis/build/buildbot/buildbot/master/docs/developer/data.rst:412: WARNING: broken link: https://github.com/buildbot/buildbot/blob/master/master/buildbot/test/interfaces/test_data_connector.py

/home/travis/build/buildbot/buildbot/master/docs/developer/plugins-publish.rst:18: WARNING: broken link: https://packaging.python.org/en/latest/tutorial.html#creating-your-own-project

/home/travis/build/buildbot/buildbot/master/docs/developer/plugins-publish.rst:19: WARNING: broken link: http://guide.python-distribute.org/index.html

/home/travis/build/buildbot/buildbot/master/docs/developer/rest.rst:1344: WARNING: broken link: http://trac.buildbot.net/ticket/4496

/home/travis/build/buildbot/buildbot/master/docs/developer/rest.rst:734: WARNING: broken link: http://trac.buildbot.net/ticket/4496

/home/travis/build/buildbot/buildbot/master/docs/developer/tests.rst:186: WARNING: broken link: https://github.com/buildbot/buildbot/blob/master/master/buildbot/fake

/home/travis/build/buildbot/buildbot/master/docs/developer/www-base-app.rst:40: WARNING: broken link: http://angular-ui.github.com/ng-grid/

/home/travis/build/buildbot/buildbot/master/docs/developer/www-base-app.rst:153: WARNING: broken link: http://trac.buildbot.net/ticket/4496

/home/travis/build/buildbot/buildbot/master/docs/manual/cfg-buildsteps.rst:1567: WARNING: broken link: http://www.microsoft.com/en-us/download/details.aspx?id=17657

/home/travis/build/buildbot/buildbot/master/docs/manual/cfg-buildsteps.rst:935: WARNING: broken link: http://www.monotone.ca/

/home/travis/build/buildbot/buildbot/master/docs/manual/cfg-buildsteps.rst:1567: WARNING: broken link: http://technet.microsoft.com/en-us/library/cc733145.aspx

/home/travis/build/buildbot/buildbot/master/docs/manual/cfg-buildsteps.rst:1885: WARNING: broken link: http://divmod.org/trac/wiki/DivmodPyflakes

/home/travis/build/buildbot/buildbot/master/docs/manual/cfg-changesources.rst:838: WARNING: broken link: https://bitbucket.org/owner/slug/

/home/travis/build/buildbot/buildbot/master/docs/manual/cfg-wwwhooks.rst:254: WARNING: broken link: https://your.gitlab.server/help/web_hooks

/home/travis/build/buildbot/buildbot/master/docs/manual/cfg-wwwhooks.rst:285: WARNING: broken link: http://gitorious.org/gitorious/pages/WebHooks

/home/travis/build/buildbot/buildbot/master/docs/manual/concepts.rst:115: WARNING: broken link: http://www.monotone.ca/

/home/travis/build/buildbot/buildbot/master/docs/manual/customization.rst:252: WARNING: broken link: http://Divmod.org

/home/travis/build/buildbot/buildbot/master/docs/relnotes/0.8.6.rst:187: WARNING: broken link: https://github.com/buildbot/buildbot/blob/master/master/contrib/fakemaster.py`

/home/travis/build/buildbot/buildbot/master/docs/relnotes/0.8.6.rst:221: WARNING: broken link: https://github.com/buildbot/buildbot/blob/master/master/docs/release-notes/

/home/travis/build/buildbot/buildbot/master/docs/relnotes/0.8.9.rst:79: WARNING: broken link: http://trac.buildbot.net/ticket/1478

make: *** [linkcheck] Error 1
```

**Comment by djmitche** [comment]
This looks good, I assume it's "DO NOT MERGE" until the links are fixed?

**Comment by rutsky** [comment]
@djmitche yes, I think links should be fixed first, and then we merge this.

I haven't fixed them because some of them actually in deprecated part of documentation which should be updated for 0.9.0 release (e.g. `https://github.com/buildbot/buildbot/blob/master/master/buildbot/status/web/template/forms.html`).

**Comment by rutsky** [comment]
@tardyp what is "bug 4496"? https://github.com/buildbot/buildbot/search?utf8=%E2%9C%93&q=4496

There is no such thing on Trac and among Buildbot pull requests.

**Comment by rutsky** [comment]
At this moment there should be only following issues:

```
sphinx-build -b linkcheck -d _build/doctrees  -q . _build/linkcheck
/home/bob/stuff/buildbot/buildbot/master/docs/developer/cls-forcesched.rst:70: WARNING: broken link: https://github.com/buildbot/buildbot/blob/mas
ter/master/buildbot/status/web/template/forms.html
/home/bob/stuff/buildbot/buildbot/master/docs/developer/rest.rst:1344: WARNING: broken link: http://trac.buildbot.net/ticket/4496
/home/bob/stuff/buildbot/buildbot/master/docs/developer/rest.rst:734: WARNING: broken link: http://trac.buildbot.net/ticket/4496
/home/bob/stuff/buildbot/buildbot/master/docs/developer/www-base-app.rst:153: WARNING: broken link: http://trac.buildbot.net/ticket/4496
```

`forms.html` should be fixed in #2145 

For `bug 4496` I hope @tardyp will tell what to do with it.

**Comment by tardyp** [comment]
its not 4496 its http://trac.buildbot.net/ticket/2673
not sure why the error

**Comment by rutsky** [comment]
@tardyp ok, thanks, I'll fix this.

**Comment by rutsky** [comment]
@tardyp what about 4496 here: https://github.com/buildbot/buildbot/blame/c21eed655f57528adf4f72118be1ac586cc2aeb5/master/docs/developer/www-base-app.rst#L154

Is it http://trac.buildbot.net/ticket/3396

**Comment by tardyp** [comment]
> Is it http://trac.buildbot.net/ticket/3396?

I believe so, but you can remove the line, as there is no more problem with npm > 5
I use it locally without issue, and the releases are generated using it

**Comment by rutsky** [comment]
@tardyp, @djmitche ping. This PR is finally ready to be reviewed/merged :)

File: master/docs/developer/cls-forcesched.rst [hunk 1]
```diff
@@ -69,7 +73,7 @@ This section contains information to help users or developers who are interested
 
            A list of types that the parameter conforms to.
            These are used by the jinja template to create appropriate html form widget.
-           The available values are visible in :src:`master/buildbot/status/web/template/forms.html` in the ``force_build_one_scheduler`` macro.
+           The available values are visible in ``master/buildbot/status/web/template/forms.html`` in the ``force_build_one_scheduler`` macro.
```
Comment (djmitche) [inline master/docs/developer/cls-forcesched.rst]: I assume the error here was that the source file doesn't exist.  I'm not sure what it should now refer to -- @tardyp ?

**Comment by djmitche** [comment]
Looks good -- ship it!

File: master/docs/developer/cls-forcesched.rst (hunk 1 above)
Comment (rutsky) [inline master/docs/developer/cls-forcesched.rst]: Yes, `forms.html` has been removed.
I opened ticket for tracking update of `ForceScheduler` docs: http://trac.buildbot.net/ticket/3477

Comment (rutsky) [inline master/docs/developer/cls-forcesched.rst]: I also added note that this docs page has outdated info at the begin of the page.

**Pull request merged**